### PR TITLE
Add reference-bundle file

### LINKED ISSUE #294: Docs needed: definitive list of bundle properties and format

### PR DESCRIPTION
**Pull request by pmatulis**
resolves #294

### CODE REVIEW

File: src/en/reference-bundle.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+Page [Charm bundles][charms-bundles] gives an overview of bundle usage.
+
+## Charm series
+
+What series a charm will use can be influenced in several ways. Some of these
+are set within the bundle file while some are not. When using bundles, the
+series is determined using rules of precedence (most preferred to least):
+
+ - the series stated in a charm URL (see `charm` under the
```
Comment (wallyworld) [inline src/en/reference-bundle.md]: We are running away as fast as possible from using charm urls with series embedded, preferring multi-series charms (with no series in the url). I'd rather not see us document a feature we are trying to get rid of.

File: src/en/reference-bundle.md [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+   `<application name>` element)
+ - the series stated for an application (see `series` under the
+   `<application name>` element)
+ - the series given by the top level `series` element
+ - the top-most series specified in a charm's `metadata.yaml` file
```
Comment (wallyworld) [inline src/en/reference-bundle.md]: And the final default is the most recent, stable LTS

File: src/en/reference-bundle.md [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+## Bundle file properties and format
+
+Comments (lines that begin with a '#' character) are used to explain various
+parts of the file. In addition, any line that is not a comment should be
+considered as an example. If such a line is duplicated it is for the sole
+purpose of showing another example. Do not duplicate lines in a real bundle
+file.
+
+A bundle file has six top level elements:
```
Comment (wallyworld) [inline src/en/reference-bundle.md]: Missing from the list below is "bundle"

File: src/en/reference-bundle.md [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+    resources:
+      easyrsa: /absolute/path/to/file
+
+    #
+    # num_units:
+    #
+    # Specifies the number of units to deploy. The default value is '0'. 
+    #
+
+    num_units: 2
```
Comment (wallyworld) [inline src/en/reference-bundle.md]: Only relevant for non-k8s bundles. "scale" is used for k8s bundle.
num_units is supported for k8s but scale is recommended and preferred

File: src/en/reference-bundle.md [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+
+    #
+    # to:
```
Comment (wallyworld) [inline src/en/reference-bundle.md]: Only relevant for non-k8s bundles.
"placement" is what's used for k8s.
"to" is supported for compatibility, but only key=value pairs

File: src/en/reference-bundle.md [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+      miner=1,nvidia.com/gpu
+    
+    #
+    # bindings:
+    #
+    # Maps endpoints to network spaces. Used to constrain relations to specific
+    # subnets in environments where machines have multiple network devices.
+    #
+
+    bindings:
```
Comment (wallyworld) [inline src/en/reference-bundle.md]: not for k8s bundes (yet)

File: src/en/reference-bundle.md [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+    
+    #
+    # plan:
+    #
+    # This is for third-party Juju support only. It sets the "managed
+    # solutions" plan for the application. The string has the format
+    # '<reseller-name>/<plan name>'.
+    #
+
+    plan: acme-support/default
```
Comment (wallyworld) [inline src/en/reference-bundle.md]: not for k8s

File: src/en/reference-bundle.md [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+# machines:
+#
+# Provides machines that have been targeted by the 'to' key under the
+# '<application name>' element. A machine is denoted by that same machine ID,
+# and must be quoted. Keys for 'constraints', 'annotations', and 'series' can
+# optionally be added to each machine. Containers are not valid machines in
+# this context.
+#
+
+machines:
```
Comment (wallyworld) [inline src/en/reference-bundle.md]: not for k8s

File: src/en/reference-bundle.md (hunk 1 above)
Comment (pmatulis) [reply src/en/reference-bundle.md]: Gotcha. I may get around to updating the help for `deploy` to make things consistent.

File: src/en/reference-bundle.md (hunk 2 above)
Comment (pmatulis) [reply src/en/reference-bundle.md]: Oh nice.

**Comment by pmatulis** [comment]
[Docs needed: definitive list of bundle properties and format](https://trello.com/c/M32jNvKE/1-docs-needed-definitive-list-of-bundle-properties-and-format)

**Pull request merged**